### PR TITLE
feat: Allow to select WASM platform when using Docker.

### LINKED ISSUE #208: Could not open '/lib64/ld-linux-x86-64.so.2'
If you're looking at this issue, it likely means you are on an M1 architecture (or at least not x86-64).

The problem is that the `prebuild-wasm` script defined in the `package.json` _is not_ platform agnostic.  The default (presumably from Docker) is to use an x86-64 architecture.  Meaning, if you are not on an x86-64 architecture you'll likely end up with an error like this:

```shell
> llhttp@8.1.0 build /home/node/llhttp
> ts-node bin/generate.ts

qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
Error: Command failed: /home/node/wasi-sdk-12.0/bin/clang  --sysroot=/home/node/wasi-sdk-12.0/share/wasi-sysroot  -target wasm32-unknown-wasi  -Ofast  -fno-exceptions  -fvisibility=hidden  -mexec-model=reactor  -Wl,-error-limit=0  -Wl,-O3  -Wl,--lto-O3  -Wl,--strip-all  -Wl,--allow-undefined  -Wl,--export-dynamic  -Wl,--export-table  -Wl,--export=malloc  -Wl,--export=free  /home/node/llhttp/build/c/*.c  /home/node/llhttp/src/native/*.c  -I/home/node/llhttp/build  -o /home/node/llhttp/build/wasm/llhttp.wasm
    at checkExecSyncError (child_process.js:616:11)
    at Object.execSync (child_process.js:652:15)
    at Object.<anonymous> (/home/node/llhttp/bin/build_wasm.ts:53:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    ...
```

## Solution
For me on an M1 mac, the solution was to change the `package.json`
from:
`    "prebuild-wasm": "docker build -t llhttp_wasm_builder . && npm run wasm -- --setup",`
to:
`    "prebuild-wasm": "docker build --platform=linux/amd64 -t llhttp_wasm_builder . && npm run wasm -- --setup",`

If you are not on an M1 mac, **you'll need to specify the correct architecture for yourself**

### PR DESCRIPTION
**Pull request by ShogunPanda**
I've added a `process.env.WASM_PLATFORM` to the build script to select the platform to build Docker image for.
Also, I've updated Docker image to be smaller and to use Node 16 to build.

Fixes #208.

### CODE REVIEW

**Comment by ShogunPanda** [comment]
@dnlup As you originally work on this, WDYT?

**Review by mcollina — APPROVED** [review]
lgtm

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,8 @@
-FROM node:14.16.0-buster
-
+FROM node:16-alpine
```
Comment (RafaelGSS) [inline Dockerfile:1]: why 16?

Comment (ShogunPanda) [reply Dockerfile:1]: 14 was using llhttp 2, so I made sure to build using the same supported version.

Moreover, 14 is going EOL in 3 months so my lazy me took control and included the change in this PR as well.

**Comment by mcollina** [comment]
I would recommend we bump it to v18, September is close too.

**Comment by ShogunPanda** [comment]
Done, updated Docker to use 18.

**Comment by ShogunPanda** [comment]
@mcollina @RafaelGSS Green light to merge this?

**Review by RafaelGSS — APPROVED** [review]
Is it a major?

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by ShogunPanda** [comment]
> Is it a major?

No, it's not. Docker is only used within the build process so it's not and user facing API.

**Comment by dnlup** [comment]
Sorry @ShogunPanda , I missed this. Great work!

**Pull request merged**